### PR TITLE
Fixed AVX detection

### LINKED ISSUE #911: Issue detecting AVX architecture on MacOS
I've run into some strange behaviour with a little test program that I'm writing, where `xsimd::available_architectures()` correctly detects that the CPU supports AVX2 instructions but does not detect that the CPU also supports AVX instructions.

I believe the CPU does in fact support AVX since when I run `sysctl -a | grep machdep.cpu`, I see:
Screen Shot 2023-03-25 at 6 56 16 PM

Another library that I use regularly has a [similar implementation](https://github.com/juce-framework/JUCE/blob/69795dc8e589a9eb5df251b6dd994859bf7b3fab/modules/juce_core/native/juce_intel_SharedCode.h#L50) that appears to give the correct result. I'd be happy to test out a potential fix, but I'm not sure I'd feel comfortable touching that part of the code myself, since I'd probably break more things than I would fix.

### PR DESCRIPTION
**Pull request by JohanMabille**
Fixes #911 

cc @serge-sans-paille

### CODE REVIEW

File: include/xsimd/config/xsimd_cpuid.hpp [hunk 1]
```diff
@@ -141,6 +141,12 @@ namespace xsimd
                 if (sse4_2)
                     best = std::max(best, fma3<xsimd::sse4_2>::version() * fma3_sse);
 
+                avx = regs[2] >> 28 & 1;
```
Comment (serge-sans-paille) [inline include/xsimd/config/xsimd_cpuid.hpp:144]: I don't understand the fix, can you explain?

Comment (JohanMabille) [reply include/xsimd/config/xsimd_cpuid.hpp:144]: regs is used before being overwritten with the call o `get_cpuid(regs, 0x80000001);`

Comment (serge-sans-paille) [reply include/xsimd/config/xsimd_cpuid.hpp:144]: ok. Maybe worth having separate locale variable for each call to `get_cpuid` then, right?

Comment (JohanMabille) [reply include/xsimd/config/xsimd_cpuid.hpp]: Indeed.

**Pull request merged**